articles/networking-models-introductory-guide/index.md

```diff
@@ -0,0 +1,114 @@
+
+* **Network Layer**: The network layer is one of the most important layers. It enables many features such as
+  * Address Assignment:  IP addresses are assigned to the host. There are two ways of assigning addresses: Static and Dynamic. Static addresses are assigned manually and do not change under any circumstances. Dynamic IP's, on the other hand, are assigned on an as-and-when-required basis. Dynamic setting
+  * Routing: Selecting the route can be done manually or automatically. Today, most of it is automatic. There are two predominant algorithms used for routing: Distance Vector Routing and Link State Routing. 
+  * Fragmentation: In the transport layer, there is a constraint on the maximum allowable size for data. Therefore, bits are segmented accordingly in the transport layer. Fragmentation is the same process applied to the segmented packets received from the transport layer. The aim is to accommodate datagrams received from the transport layer into frames. 
+   
+* **Data Link Layer**: The main responsibility of Data Link Layer(DLL) is ensuring Flow Control, Error Control, Access Control, Framing, and reading physical address. We will address each of the processes in detail:
+  * Framing: The process of taking a packet from the above layer and adding a frame to the packet is called framing. The frame includes data such as the end of the packet, message length, etc to achieve accurate information at the receiving end. 
+  * Flow Control: DLL restricts the size of the traffic and waits for the receiver to acknowledge the first batch of frames before sending the next batch.
+  * Error Control: Due to long-distance transmission, sometimes the bits might get corrupted. The corruption of bits leads to poor service. Therefore, there are various approaches to handling data corruption.
+    * Discarding the data corruption bits
+    * Repairing the corrupted bits
+  * There are various error correction algorithms like Cyclic Redundancy Check, Checksum, Parity Bits, etc. 
+
+* **Physical Layer**: This layer deals with electrical, mechanical, functional, and procedural characteristics of physical links. Network topology comes under this layer. One prominent aspect of the physical layer is encoding. Encoding refers to the representation of data. The objective of encoding is to ensure maximum probability that the message is transmitted without any errors. There are different types of encoding available. They differ in the way the 0's and 1's are represented. Basic representation is -5V signal for 0 and +5V signal for 1. The probability of error is high, and therefore various other schemes are required. Let's review some of these visually. 
+    * Manchester Encoding: ![Source: https://manikareahome.files.wordpress.com/](https://manikareahome.files.wordpress.com/2019/09/differential-manchester.png?w=1001)
```

Comment:
@lalith1403  these two images should be hosted within the repo - if you could add them as files to the article 🚀